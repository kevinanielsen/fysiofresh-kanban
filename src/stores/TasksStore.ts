import { defineStore } from "pinia";

export const useTasksStore = defineStore("tasks", {
  state: () => ({
    todo: [{ id: 0, title: "Do the dishes", description: "" }],
    doing: [],
    done: [],
  }),
  actions: {
    addTask(title: string, description: string) {
      const id = Math.round(Math.random() * 10000);
      this.todo.push({ id, title, description });
    },
  },
});
