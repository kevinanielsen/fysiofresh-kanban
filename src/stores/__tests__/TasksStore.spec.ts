import { it, expect, beforeEach, describe } from "vitest";
import { useTasksStore } from "../TasksStore";
import { createPinia, setActivePinia } from "pinia";

beforeEach(() => {
  setActivePinia(createPinia());
})

describe("tasksStore", () => {
  it('should ensure default state', () => {
    //Given a new store without any changes
    //When store is created
    //Then the default state is

    //Arrange
    const tasksStore = useTasksStore();

    //Act - we're testing the constructor, there is no act.

    //Assert
    expect(tasksStore.todo).toMatchInlineSnapshot(`
      [
        {
          "description": "",
          "id": 0,
          "title": "Do the dishes",
        },
      ]
    `);
  });
  it("should add a task", () => {
    //Given a new store without any changes
    //When a task is added
    //Then the task is added to the todo list

    //Arrange
    const tasksStore = useTasksStore();

    //Act
    tasksStore.addTask("Do the dishes", "Clean the dishes");

    //Assert
    expect(tasksStore.todo[tasksStore.todo.length - 1].description).toMatchInlineSnapshot(`"Clean the dishes"`);
    expect(tasksStore.todo[tasksStore.todo.length - 1].title).toMatchInlineSnapshot(`"Do the dishes"`);
  });
})
