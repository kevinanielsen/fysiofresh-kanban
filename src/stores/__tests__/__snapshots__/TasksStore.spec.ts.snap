// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tasksStore > should add a task 1`] = `
[
  {
    "description": "",
    "id": 0,
    "title": "Do the dishes",
  },
  {
    "description": "Clean the dishes",
    "id": 2293,
    "title": "Do the dishes",
  },
]
`;

exports[`tasksStore > should ensure default state 1`] = `
{
  "$dispose": [Function],
  "$id": "tasks",
  "$onAction": [Function],
  "$patch": [Function],
  "$reset": [Function],
  "$subscribe": [Function],
  "_customProperties": Set {},
  "_hmrPayload": {
    "actions": {
      "addTask": [Function],
    },
    "getters": {},
    "hotState": RefImpl {
      "__v_isRef": true,
      "__v_isShallow": false,
      "_rawValue": {},
      "_value": {},
      "dep": undefined,
    },
    "state": [
      "todo",
      "doing",
      "done",
    ],
  },
  "_hotUpdate": [Function],
  "_p": {
    "_a": null,
    "_e": EffectScope {
      "_active": true,
      "cleanups": [],
      "detached": true,
      "effects": [],
      "parent": undefined,
      "scopes": [
        EffectScope {
          "_active": true,
          "cleanups": [],
          "detached": false,
          "effects": [],
          "index": 0,
          "parent": [Circular],
        },
      ],
    },
    "_p": [],
    "_s": Map {
      "tasks" => [Circular],
    },
    "install": [Function],
    "state": RefImpl {
      "__v_isRef": true,
      "__v_isShallow": false,
      "_rawValue": {
        "tasks": {
          "doing": [],
          "done": [],
          "todo": [
            {
              "description": "",
              "id": 0,
              "title": "Do the dishes",
            },
          ],
        },
      },
      "_value": {
        "tasks": {
          "doing": [],
          "done": [],
          "todo": [
            {
              "description": "",
              "id": 0,
              "title": "Do the dishes",
            },
          ],
        },
      },
      "dep": undefined,
    },
    "use": [Function],
  },
  "addTask": [Function],
  "doing": [],
  "done": [],
  "todo": [
    {
      "description": "",
      "id": 0,
      "title": "Do the dishes",
    },
  ],
}
`;
